### PR TITLE
Simplify MediaServerConfiguration code

### LINKED ISSUE #28025: Multistore - Advanced Parameters > Performance:media - use ShopConstraint as third argument
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md)
- [X] I have already [searched in existing features request](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3AImprovement%2CFeature%2CDeveloper+Feature+) and found no previous suggestion of this feature.

### Is your feature request related to a problem?

no

### Describe the solution you'd like

Simplifies the number of conditionals branches. This makes the code easier to maintain. Use ShopConstraint and added unit test.

It's about code refactoring over configuring media servers and code behind it. I took one part of the code and improved it.

https://github.com/PrestaShop/PrestaShop/pull/27660#discussion_r818547416

### Alternatives you've considered

_No response_

### Additional context

I have added unit test in the PR

### Do you plan to work on this feature?

- [ ] I'm willing to contribute a [formal specification](https://github.com/PrestaShop/prestashop-specs).
- [ ] I'm willing to provide any wireframes or design assets required for this feature.
- [X] I'm willing to submit a Pull Request that implements this feature.
- [ ] I'm willing to help verify that the implemented feature works as intended and produces no unintended side effects.

### PR DESCRIPTION
**Pull request by davidglezz**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Simplifies the number of conditionals branches. This makes the code easier to maintain. Use ShopConstraint and added unit test.
| Type?             | refacto
| Category?         | CO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Fixes #28025
| How to test?      | CI green, I have added unit tests
| Possible impacts? | no

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/27660)

### CODE REVIEW

**Review by matthieu-rolland — DISMISSED** [review]

File: src/Adapter/Media/MediaServerConfiguration.php
Comment (Progi1984) [inline src/Adapter/Media/MediaServerConfiguration.php]:
```suggestion
    private function isValidDomain(string $domainName): bool
```

**Review by Progi1984 — COMMENTED** [review]
A small feedback and it's good for me.

**Comment by davidglezz** [comment]
I committed your suggestion. @Progi1984

**Comment by davidglezz** [comment]
Sorry, I made a rebase and lost @matthieu-rolland approval

**Review by matks — DISMISSED** [review]

File: src/Adapter/Media/MediaServerConfiguration.php
Comment (atomiix) [inline src/Adapter/Media/MediaServerConfiguration.php]: Maybe time to use `ShopConstraint` as third argument as it's now deprecated to omit it? 🙃

Comment (davidglezz) [reply src/Adapter/Media/MediaServerConfiguration.php]: @atomiix Could you tell me how I can do it? I have searched the prestashop source code and of the more than 100 uses of configuration, 0 use the 3rd parameter $shopConstraint.
and I see that all the ones I have found use `ConfigurationInterface` instead of `ShopConfigurationInterface`

Comment (Progi1984) [reply src/Adapter/Media/MediaServerConfiguration.php]: @davidglezz You can find some informations in these issue : #25058 (and check all connected PR)

Comment (matthieu-rolland) [reply src/Adapter/Media/MediaServerConfiguration.php]: I think it's ok not to do it in this PR...

It would mean extending `AbstractMultistoreConfiguration` and in the method `updateCondiguration` you would have to get the shop constraint, like this:

`$shopConstraint = $this->getShopConstraint();`

Then to update the configuration values with the new `updateConfigurationValue` method:

`$this->updateConfigurationValue(...)`

But you would also have to add a new `buildResolver` method (which is mandatory to respect the contract of the parent abstract class)

We usually do all of this when we migrate a form in multistore, but for now, it seems overkill and out of the scope of this PR...   

If you want to do this though, you can look at `src/Adapter/Shop/MaintenanceConfiguration.php`

Comment (matthieu-rolland) [reply src/Adapter/Media/MediaServerConfiguration.php]: @davidglezz so you can tell us, if you want to do it we can help, otherwise your PR can be accepted as is 👍

**Review by Progi1984 — COMMENTED** [review]
Could you fix PHPStatn errors on `src/Adapter/Media/MediaServerConfiguration.php` ?

File: src/Adapter/Media/MediaServerConfiguration.php
Comment (davidglezz) [reply src/Adapter/Media/MediaServerConfiguration.php]: Done, thanks for the help

**Comment by davidglezz** [comment]
Done ✅ 
- use ShopConstraint
- Added unit test
- Rebased

**Review by Progi1984 — DISMISSED** [review]

**Review by atomiix — DISMISSED** [review]
Nice work @davidglezz!

**Review by AureRita — DISMISSED** [review]
Hi @davidglezz,

Thank you for you pr ! Can you create an issue to allow the QA team to follow all the steps to test your pr, without this, the QA team can't test your PR

**Comment by davidglezz** [comment]
@AureRita I have created an issue.

**Comment by Robin-Fischer-PS** [comment]
@davidglezz Can this PR be tested functionally ? if yes, please update your `how to test` or `the issue` so we can test it.
if not, maybe it's QA By dev ?
@PrestaShop/prestashop-core-developers With no clear how to test or issue, this PR should not be `waiting for QA`

**Comment by kpodemski** [comment]
Hey @davidglezz 

I think a regular testing campaign to test media server configuration should be enough? Setting new values, deleting, etc.? If so, please update the description accordingly so we can have it tested by unit tests + manually before merging this PR 💪

**Comment by davidglezz** [comment]
This PR does not change the behavior, its only a refactor, so it can be tested by the new unit tests (added in this PR) + manually or, if exist, automated test as it was done before.

> please update the description accordingly so we can have it tested by unit tests + manually before merging this PR 

@kpodemski  I'm sorry I don't know what I can put in the description, any suggestion or help?

**Review by kpodemski — DISMISSED** [review]

**Comment by kpodemski** [comment]
@PrestaShop/qa-team 

Do we have any automated tests for this page? If so, it would be good to run it. If not, tests of media servers configuration should be enough

**Comment by HanaRebaiQA** [comment]
Hello @kpodemski 

Yes! we can launch the automated tests in this PR. But, i noticed that the related issue is with status `Invalid`
I think that the issue should be `Verified` to can test the PR

Waiting for your feedback!

Thank you!

**Comment by davidglezz** [comment]
I don't know what else I can do.
I just wanted to improve the code a bit.

**Comment by hibatallahAouadni** [comment]
Hello @davidglezz 

> We need a detailed description about the work you have done it in your PR :wink: 
> > this is an issue for the QA team to test a PR
> > I have added unit test in the PR
> 
> It's not a good description of the proposed solution nor for additional context :sweat_smile: 
> So, please edit and notify me :wink: 

You need to reply my message on the related issue :pray: 
Waiting for your feedback.

Thanks!

**Pull request closed without merging**
